### PR TITLE
feat: wallet server open api and CHAPI bridge interfaces

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- open api support for wallet server
- CHAPI bridge interfaces

TODO:
- agent support
- refactoring webwallet docker-compose to segregate UI & server
components

- Part of #636
- Part of #635
- Part of #637

Signed-off-by: sudesh.shetty <sudesh.shetty@securekey.com>

### CODE REVIEW

**Review by fqutishat — APPROVED** [review]

File: pkg/restapi/wallet/chapibridge/operations.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	RequestCHAPIAppProfile = "/request-app-profile"
+	SendCHAPIRequest       = "/send-chapi-request"
+)
+
+// Operation is REST service operation controller for CHAPI bridge features.
+type Operation struct {
+	handlers []rest.Handler
+}
+
+// Provider describes dependencies for this command.
```
Comment (DRK3) [inline pkg/restapi/wallet/chapibridge/operations.go:30]: Just to double check: Is this really the right comment?

Comment (sudeshrshetty) [reply pkg/restapi/wallet/chapibridge/operations.go:30]: Yes, dependencies from context

File: pkg/restapi/wallet/chapibridge/operations.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+type Provider interface {
+}
+
+// New returns new CHAPI bridge rest controller instance.
```
Comment (DRK3) [inline pkg/restapi/wallet/chapibridge/operations.go:34]: Small note: You use upper case REST in your other comments - I suggest picking one style and sticking to it (preferably upper case)

File: pkg/restapi/wallet/chapibridge/operations.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+func New(ctx Provider) (*Operation, error) {
+	o := &Operation{}
+
+	o.registerHandler()
+
+	return o, nil
+}
+
+// GetRESTHandlers get all controller API handler available for this protocol service.
```
Comment (DRK3) [inline pkg/restapi/wallet/chapibridge/operations.go:43]: should be `gets`

File: pkg/restapi/wallet/chapibridge/operations.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+// SendChapiRequest swagger:route POST /wallet/send-chapi-request chapi-bridge chapiRequest
+//
+// Sends CHAPI request to given wallet application ID.
+// Response contains CHAPI request.
+//
+// Responses:
+//    default: genericError
+//    200: chapiResponse
+func (c *Operation) SendChapiRequest(rw http.ResponseWriter, req *http.Request) {
```
Comment (DRK3) [inline pkg/restapi/wallet/chapibridge/operations.go:94]: May want to capitalize CHAPI

File: pkg/restapi/wallet/controller.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+// GetRESTHandlers returns all REST handlers provided by wallet controller.
+func GetRESTHandlers(ctx *context.Provider) ([]rest.Handler, error) { //nolint:interfacer
+	// chapiBridge rest controller operations,
+	chapiBridge, err := chapibridge.New(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// creat handlers from all REST operations.
```
Comment (DRK3) [inline pkg/restapi/wallet/controller.go:24]: typo in "create"

**Review by DRK3 — APPROVED** [review]

**Review by talwinder50 — APPROVED** [review]

**Pull request merged**